Frontend/ui-driver/src/Screens/DriverEarningsScreen/View.purs

```diff
@@ -0,0 +1,1583 @@
+                    liftFlowBT $ push $ UpdateRidesEver anyRidesAssignedEver
+                  (GetRidesHistoryResp rideHistoryResponse) <- Remote.getRideHistoryReqBT "100" "0" "false" "null" (convertUTCtoISC initialState.props.date "YYYY-MM-DD")
+                  liftFlowBT $ push $ RideHistoryAPIResponseAction rideHistoryResponse.list
+                  let currentDate = getcurrentdate ""
+                      datesList = getDatesList currentDate initialState
+                  (GetRidesSummaryListResp rideSummaryResp) <- Remote.getRideSummaryListReqBT datesList
+                  liftFlowBT $ push $ RideSummaryAPIResponseAction rideSummaryResp.list currentDate datesList
+                ST.YATRI_COINS_VIEW -> do
+                  coinTransactionRes <- Remote.getCoinTransactionsReqBT initialState.props.date
+                  liftFlowBT $ push $ CoinTransactionResponseAction coinTransactionRes
+                  pure unit
+                ST.USE_COINS_VIEW -> do
+                  coinUsageHistoryRes <- Remote.getCoinUsageHistoryReqBT ""
+                  liftFlowBT $ push $ CoinUsageResponseAction coinUsageHistoryRes
+                  pure unit
+                _ -> do
+                  liftFlowBT $ push $ FaqViewAction 
+                  pure unit
+            pure $ pure unit
+        )
+  ]
+  , eval : (\action state -> do 
+    let _ = spy "DriverEarningsScreenState action" action
+    let _ = spy "DriverEarningsScreenState state" state 
+    eval action state)
+  }
+
+getDatesList :: String -> ST.DriverEarningsScreenState -> Array String
+getDatesList todaysDate state = do
+  let storedRideSummaryData = fromMaybe [] (fetchWeekyEarningData RIDE_SUMMARY_DATA)
+  if length storedRideSummaryData > 0 then map (\x -> convertUTCtoISC x.utcDate "YYYY-MM-DD") case last storedRideSummaryData of 
+                            Just x -> getPastDays $ (differenceBetweenTwoUTC todaysDate x.rideDate)/(24 * 60 * 60)
+                            Nothing -> getPastDays (22 + getDayOfWeek (getDayName todaysDate))
+  else map (\x -> convertUTCtoISC x.utcDate "YYYY-MM-DD") (getPastDays (22 + getDayOfWeek (getDayName todaysDate)))
+
+view :: forall w . (Action -> Effect Unit) -> ST.DriverEarningsScreenState -> PrestoDOM (Effect Unit) w
+view push state =
+   relativeLayout
+    [ height MATCH_PARENT
+    , width MATCH_PARENT
+    , orientation VERTICAL
+    , onBackPressed push (const BackPressed)
+    , afterRender push (const AfterRender)
+    , background Color.white900
+    ][ frameLayout
+        [ height MATCH_PARENT
+        , width MATCH_PARENT
+        , orientation VERTICAL
+        , background Color.blue600
+        ][Anim.screenAnimationFadeInOut
+          $ linearLayout
+            [ height MATCH_PARENT
+            , width MATCH_PARENT
+            , orientation VERTICAL
+            , visibility if state.props.showShimmer then GONE else VISIBLE
```

Comment:
use boolToVisibility function from Mobility.Prelude for this purpose, wherever it is applicable